### PR TITLE
PDA Messenger DEPT is now Collapsible - and stringbuild removal

### PR DESCRIPTION
**Pull request by MarkNais**
[CHORE][FEATURE]
## About the PR 
Makes the DEPT sub-list at the top of the messenger list collapsible. Starts expanded.
Removes a bunch of string building from the PDA OS
Tested.

## Why's this needed? 
#1361 and #1362 will add two more lines to the DEPT sub-list. Gonna get crowded, and the PDA dialog window isn't scalable.
Minor performance improvement.

image
image

## Changelog

```
(u)MarkNstein:
(+)PDA DEPT list is now collapsible.
```

### CODE REVIEW

File: code/obj/item/device/pda2/base_os.dm [hunk 1]
```diff
@@ -109,25 +110,25 @@
...
 							. += "<a href='byond://?src=\ref[src];note_func=lock'>Lock</a><br>"
 						else
-							. += "<a href='byond://?src=\ref[src];input=note'>Edit</a>"
-							. += " | <a href='byond://?src=\ref[src];note_func=new'>New File</a>"
-							. += " | <a href='byond://?src=\ref[src];note_func=save'>Save</a>"
-							. += " | <a href='byond://?src=\ref[src];note_func=switchmenu'>Load</a><br>"
+							. += {"<a href='byond://?src=\ref[src];input=note'>Edit</a>
+							 | <a href='byond://?src=\ref[src];note_func=new'>New File</a>
+							 | <a href='byond://?src=\ref[src];note_func=save'>Save</a>
+							 | <a href='byond://?src=\ref[src];note_func=switchmenu'>Load</a><br>"}
```
Comment (ZeWaka) [inline code/obj/item/device/pda2/base_os.dm:116]: unsure if the spaces here will be respected

Comment (MarkNais) [reply code/obj/item/device/pda2/base_os.dm:116]: Good point, thankfully it does! 
image

File: code/obj/item/device/pda2/base_os.dm [hunk 2]
```diff
@@ -154,8 +154,8 @@
 
 						if (src.message_on)
 							if(expand_departments_list)
-								for (var/department_id in page_departments)
 								. += "<br><a href='byond://?src=\ref[src];toggle_departments_list=1;refresh=1'>*Collapse DEPT list*</a>"
```
Comment (UrsulaMejor) [inline code/obj/item/device/pda2/base_os.dm:157]:
```suggestion
								. += "<a href='byond://?src=\ref[src];toggle_departments_list=1;refresh=1'>*Collapse DEPT list*</a>"
```

Comment (UrsulaMejor) [reply code/obj/item/device/pda2/base_os.dm:157]: literally unplayable

**Pull request merged**